compiler/rustc_trait_selection/src/error_reporting/infer/mod.rs

```diff
@@ -359,15 +369,35 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
                     }
                 }
                 if let Some(ty::error::ExpectedFound { found, .. }) = exp_found
-                    && ty.boxed_ty() == Some(found)
-                    && let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(span)
+                    && let Ok(mut snippet) = self.tcx.sess.source_map().span_to_snippet(span)
```

Comment:
I know it's preexisting, but this snippet logic is unnecessary. You should be able to just give a suggestion like `"*"` that is inserted before the match scrutinee with `span.shrink_to_lo()`.

Reply:
It may also look better if you change this into a "verbose" suggestion with `span_suggestion_verbose`.

Reply:
I've gone back on this, because I'm not sure how to override a span in a suggestion using multipart. Please let me know if there's a way to do so.